src/ucs/stats/stats.h

```diff
@@ -14,15 +15,81 @@
 
 #include <ucs/sys/compiler_def.h>
 #include <ucs/sys/stubs.h>
+#include <ucs/stats/stats_fwd.h>
 
 BEGIN_C_DECLS
 
 /** @file stats.h */
 
+/* The maximum size of a class name */
+#define UCS_STATS_CLASS_NAME_MAX 50
+
+/* Realloc size (i.e. how many elements to add during realloc()) */
+#define UCS_STATS_AGG_SUM_COUNTERS_REALLOC_SZ 32
+
+
+typedef struct {
+    /* Class name */
+    char     class_name[UCS_STATS_CLASS_NAME_MAX];
+
+    /* Class base offset in the counters aggregate-sum list */
+    unsigned aggregate_sum_class_offset;
+
+} ucs_stats_aggrgt_sum_clsid_to_name_db_t;
+
+
+typedef struct {
+    /* Counter class name */
+    const char *class_name;
+
+    /* Counter name */
+    const char *name;
+
+} ucs_stats_aggrgt_sum_counter_name_t;
+
+
 void ucs_stats_init();
 void ucs_stats_cleanup();
 void ucs_stats_dump();
 int ucs_stats_is_active();
+
+/**
+ * A UCX statistics API function to return the aggregate-sum of all counters
+ * from all workers/interfaces for a compact statistics trace.
+ * Used for Score-P UCX plugin to collect a compact trace of the counters.
+ *
+ * @param [out] aggregate_sum_counters      The aggregate-sum of all counters of each class/type.
+ * @param [in]  aggregate_sum_counters_size The number of elements in aggregate_sum_counters.
+ * @param [out] clsid_to_name_db            The conversion database from clsid to name.
+ *
+ * Complexity: O(n) / recursive.
+ *
+ * Returns: The actual number of aggregate_sum counters in the output
+ *          buffer.
+ *
+ * Remarks: The function calls a recursive function.
+ */
+size_t ucs_stats_aggregate(
+        ucs_stats_counter_t *aggregate_sum_counters,
+        size_t aggregate_sum_counters_size,
+        const ucs_stats_aggrgt_sum_clsid_to_name_db_t **clsid_to_name_db);
+
+/**
+ * A UCX statistics API to get the names of counters on the last call to
+ * ucs_stats_aggregate_sum()
+ *
+ * @param [out] names_db The aggregate-sum counters names database
+ * @param [out] size     number of counters in database
+ *
+ * Complexity: O(1)
+ *
+ * Returns: None.
```

Comment:
1. use doxygen tag `@return`
2. pls document ucs_stats_aggregate() should be called before this function

Reply:
Fixed.